tests/CppUTestExt/MockSupportTest.cpp

```diff
@@ -32,85 +32,67 @@
 #include "CppUTestExt/MockFailure.h"
 #include "MockFailureTest.h"
 
-TEST_GROUP(MockSupportTest)
+TEST_GROUP(MockSupportBasicTest)
 {
-    MockExpectedCallsList *expectationsList;
-
-    void setup()
-    {
-        mock().setMockFailureStandardReporter(MockFailureReporterForTest::getReporter());
-        expectationsList = new MockExpectedCallsList;
-    }
-
-    void teardown()
-    {
-        mock().checkExpectations();
-        CHECK_NO_MOCK_FAILURE();
-        expectationsList->deleteAllExpectationsAndClearList();
-        delete expectationsList;
-        mock().setMockFailureStandardReporter(NULL);
-    }
-
-    MockCheckedExpectedCall* addFunctionToExpectationsList(const SimpleString& name)
-    {
-        MockCheckedExpectedCall* newCall = new MockCheckedExpectedCall;
-        newCall->withName(name);
-        expectationsList->addExpectedCall(newCall);
-        return newCall;
-    }
-
-    MockCheckedExpectedCall* addFunctionToExpectationsList(const SimpleString& name, int order)
-    {
-        MockCheckedExpectedCall* newCall = new MockCheckedExpectedCall;
-        newCall->withName(name);
-        newCall->withCallOrder(order);
-        expectationsList->addExpectedCall(newCall);
-        return newCall;
-    }
-
+  void teardown()
+  {
+    mock().checkExpectations();
+  }
 };
 
-TEST(MockSupportTest, clear)
+TEST(MockSupportBasicTest, clear)
 {
     mock().expectOneCall("func");
     mock().clear();
     CHECK(! mock().expectedCallsLeft());
 }
 
-TEST(MockSupportTest, checkExpectationsDoesntFail)
-{
-    mock().checkExpectations();
-}
-
-TEST(MockSupportTest, checkExpectationsClearsTheExpectations)
+TEST(MockSupportBasicTest, checkExpectationsDoesntFail)
 {
-    addFunctionToExpectationsList("foobar");
-    MockExpectedCallsDidntHappenFailure expectedFailure(mockFailureTest(), *expectationsList);
-
-    mock().expectOneCall("foobar");
     mock().checkExpectations();
-
-    CHECK(! mock().expectedCallsLeft());
-    CHECK_EXPECTED_MOCK_FAILURE(expectedFailure);
 }
 
-TEST(MockSupportTest, exceptACallThatHappens)
+TEST(MockSupportBasicTest, exceptACallThatHappens)
 {
     mock().expectOneCall("func");
     mock().actualCall("func");
     CHECK(! mock().expectedCallsLeft());
 }
 
-TEST(MockSupportTest, exceptACallInceasesExpectedCallsLeft)
+TEST(MockSupportBasicTest, exceptACallInceasesExpectedCallsLeft)
 {
     mock().expectOneCall("func");
     CHECK(mock().expectedCallsLeft());
     mock().clear();
 }
 
+TEST_GROUP(MockSupportTest)
+{
+  MockExpectedCallsListForTest expectations;
+  MockFailureReporterInstaller failureReporterInstaller;
+
+  void teardown()
+  {
+    mock().checkExpectations();
+    CHECK_NO_MOCK_FAILURE();
+  }
+};
+
```

Comment:
Okay, that's less stuff. Do you propose duplicating this in every TEST_GROUP, or stuffing it into a fixture? Duplicate the fixutre in each file, or use a header?

(Same would wokr with TEST_BASE, of course :P)